### PR TITLE
Changes power gloves to a bio chip, buffs it.

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Power gloves are now a bio chip.
It now uses the power in the grid, rather than the surplus, making it a bit easier to use when one is not in the engineering department.
It now targets missed clicks better to hit mobs, over random objects / mobs chosen randomly.
You can toggle it off, so you can open your inventory better or still disarm / punch people.

## Why It's Good For The Game

Easier to use is good.
Works with modsuits now.
Better in the bundle as it no longer relies on engineering overcharging the grid, still works if the grid is balanced like other shocks.

## Testing

Shocked skrells and beepsky.

## Changelog
:cl:
tweak: Changes power gloves to a bio chip, now works with modsuits.
tweak: Power bio chip will now target mobs better on a missed click.
tweak: Power bio chip now uses grid power vs grid surplus.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/implants/implant_shock.dm [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/obj/item/implant/shock
+	name = "power bio-chip"
+	desc = "A shockingly effective bio-chip for stunning or killing all those in your way. Do it."
+	icon_state = "lighting_bolt"
+	item_color = "r"
+	origin_tech = "combat=5;magnets=3;biotech=4;syndicate=2"
+	implant_data = /datum/implant_fluff/shock
+	implant_state = "implant-syndicate" //
```
Comment (Henri215) [inline code/game/objects/items/weapons/implants/implant_shock.dm:8]:
```suggestion
	implant_state = "implant-syndicate"
```

File: code/game/objects/items/weapons/implants/implant_shock.dm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	var/enabled = TRUE
+	var/old_mclick_override
+	var/datum/middleClickOverride/shock_implant/mclick_override = new /datum/middleClickOverride/shock_implant
+	var/last_shocked = 0
+	var/shock_delay = 3 SECONDS
+	var/unlimited_power = FALSE // Does this really need explanation?
```
Comment (Henri215) [inline code/game/objects/items/weapons/implants/implant_shock.dm:14]: 😆

File: code/_onclick/click_override.dm [hunk 3]
```diff
@@ -73,26 +75,31 @@
 		beam_from.Beam(target_atom, icon_state = "lightning[rand(1, 12)]", icon = 'icons/effects/effects.dmi', time = 6)
 		if(isliving(target_atom))
 			var/mob/living/L = target_atom
-			var/surplus_power = C.get_surplus()
+			var/powergrid = C.powernet.available_power //We want avalible power, so the station being conservative doesn't mess with glove / dark bundle users
```
Comment (Henri215) [inline code/_onclick/click_override.dm:78]:
```suggestion
			var/powergrid = C.powernet.available_power //We want available power, so the station being conservative doesn't mess with glove / dark bundle users
```

File: code/game/objects/items/weapons/implants/implant_shock.dm [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	var/shock_range = 7
+
+
+/obj/item/implant/shock/activate()
+	enabled = !enabled
+	to_chat(imp_in, "You toggle the implant [enabled? "on" : "off"]")
```
Comment (Henri215) [inline code/game/objects/items/weapons/implants/implant_shock.dm:20]:
```suggestion
	to_chat(imp_in, "<span class='notice'>You toggle the implant [enabled? "on" : "off"].</span>")
```
Oh and.. we call it implant or bio-chip here? I got no idea :D

**Comment by 1080pCat** [comment]
> Shocked skrells and beepsky.

bro what'd beepsky do to you

File: code/game/objects/items/weapons/implants/implantfluff.dm [hunk 5]
```diff
@@ -134,3 +134,9 @@
 	life = "Unknown. No collected sample has been active long enough to determine lifespan."
 	notes = "Manually activated by the user."
 	function = "Allows the user to summon a box from a bluespace pocket located inside the implant. The exterior of this box is lined with experimental cloaking panels which render the box invisible to the naked eye."
+
+/datum/implant_fluff/shock
+	name = "Syndicate ARC-7 \"Power\" Bio-chip"
+	life = "5-12 months, depending on active use."
+	notes = "Manually activated by the user."
+	function = "An effective biochip designed to redirect large amounts of power in destructive arcs. Attempts to reduce the minimum power drain to allow for weapon and cell charging have failed so far."
```
Comment (Burzah) [inline code/game/objects/items/weapons/implants/implantfluff.dm:142]:
```suggestion
	function = "An effective biochip designed to redirect large amounts of power in destructive arcs. Attempts to reduce the minimum power drain, for weapon and cell charging, have failed so far."
```
Totally not required, but I think something like this flows a bit better.

Comment (Daylight2) [reply code/game/objects/items/weapons/implants/implantfluff.dm:142]: I like the original version better. The commas seem unnecessary - the sentence flows better without them.

**Review by DGamerL — CHANGES_REQUESTED** [review]
Needs a merge master and to have the mentions of implant replaced with `bio_chip` and such

File: code/_onclick/click_override.dm [hunk 6]
```diff
@@ -73,26 +75,31 @@
 		beam_from.Beam(target_atom, icon_state = "lightning[rand(1, 12)]", icon = 'icons/effects/effects.dmi', time = 6)
 		if(isliving(target_atom))
 			var/mob/living/L = target_atom
-			var/surplus_power = C.get_surplus()
+			var/powergrid = C.powernet.available_power //We want available power, so the station being conservative doesn't mess with glove / dark bundle users
```
Comment (Sirryan2002) [inline code/_onclick/click_override.dm:78]:
```suggestion
			var/powergrid = C.powernet.get_available_power() //We want available power, so the station being conservative doesn't mess with glove / dark bundle users
```
getter with safety checks

**Review by Sirryan2002 — APPROVED** [review]
one thing, otherwise this looks good

File: code/_onclick/click_override.dm [hunk 7]
```diff
@@ -73,26 +75,31 @@
 		beam_from.Beam(target_atom, icon_state = "lightning[rand(1, 12)]", icon = 'icons/effects/effects.dmi', time = 6)
 		if(isliving(target_atom))
 			var/mob/living/L = target_atom
-			var/surplus_power = C.get_surplus()
+			var/powergrid = C.get_available_power() //We want available power, so the station being conservative doesn't mess with glove / dark bundle users
```
Comment (Sirryan2002) [inline code/_onclick/click_override.dm:78]: I just realized that this is wrong, oops

Comment (Qwertytoforty) [reply code/_onclick/click_override.dm:78]: yeah just made the change

File: code/game/objects/items/weapons/bio_chips/bio_chip_shock.dm [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+	implant_data = /datum/implant_fluff/shock
+	implant_state = "implant-syndicate"
+	var/enabled = TRUE
+	var/old_mclick_override
+	var/datum/middleClickOverride/shock_implant/mclick_override = new /datum/middleClickOverride/shock_implant
+	var/last_shocked = 0
+	var/shock_delay = 3 SECONDS
+	var/unlimited_power = FALSE // Does this really need explanation?
+	var/shock_range = 7
+
```
Comment (DGamerL) [inline code/game/objects/items/weapons/bio_chips/bio_chip_shock.dm:16]:
```suggestion
```

**Review by DGamerL — APPROVED** [review]
If the power stuff is sorted with Sirryan, LGTM

**Review by S34NW — APPROVED** [review]

**Pull request merged**